rfc/2022/2022-09-15_Design-System-Team_Migrate-vets-website-off-of-formation.md

```diff
@@ -0,0 +1,106 @@
+# RFC: formation and USWDS1 will be removed from `vets-website` and `content-build` and will instead use the Design System's new `css-library` based on USWDS3
+
+* Comment Deadline: `2022-10-11`
+* Team Crew and Name: Platform/DST
+* Authors:
+  * [Brooks Johnson](https://github.com/bkjohnson)
+  * [Contributing Author 1 Github Handle](https://github.com/ContributingAuthor1Handle)
+  * [Contributing Author 2 Github Handle](https://github.com/ContributingAuthor2Handle)
+* [Original RFC Pull Request](https://github.com/department-of-veterans-affairs/va.gov-platform-arch/pull/-1)
+
+## Background
+
+`formation`, the CSS framework for va.gov, is dependent on USWDS version 1 as is the rest of va.gov. This is now a dead branch and the latest version of USWDS is version 3.
+In order to continue to receive updates from USWDS, we need to switch to a more recent version.
+
+## Motivation
+
+The Design System team is developing a new CSS library which is based on USWDS3 and may contain additional VA-specific styling.
+By relying on USWDS internals, we are able to provide a lot of utilities while having to maintain less SASS ourselves.
+We also want to namespace all of the classes we provide rather than allow developers full unlimited access to everything within USWDS.
+This gives the Design System more control and will also make future maintenance and updates easier.
+
+We also hope to be able to improve performance with this new css library in a number of ways, including:
+
+- smaller stylesheets due to elimination of `!important` declaration
+- smaller stylesheets due to simpler utility names (`vads-u-background-color--gray` => `vads-bg-gray`)
+
+(Note: these name changes are why teams will need to update their apps to use the names of the new classes)
+
```

Comment:
Perhaps we could point or include the work you did here to estimate the potential performance size improvement by moving to shorter class names? Quantifying the performance improvements, even if it's hand-wavy, would be great here.

Reply:
https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/48/commits/8da0d04f0bc661ec0f03aac682f05ff3a7b49186